### PR TITLE
Implemented Palindrome Partitioning using Backtracking algorithm

### PR DESCRIPTION
**Pull request by Trikcode**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] I have added  Palindrome Partitioning, a new  algorithm which involves partitioning a string into substrings such that every substring is a palindrome.

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by appgurueu — CHANGES_REQUESTED** [review]
* This should not be a class, but rather a single function. You can nest helper functions ("closures") within that function if it is useful.
* `isPalindrome` should be imported rather than implemented again.
* I believe this should probably be called "recursive" rather than "backtracking".

File: Recursive/PalindromePartitioning.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Problem Statement: Given a string s, partition s such that every substring of the partition is a palindrome. Return all possible palindrome partitioning of s.
+ * what is palindrome partitioning?
+ * - Palindrome partitioning means, partitioning a string into substrings such that every substring is a palindrome.
+ * Reference to know more about palindrome partitioning:
+ * - https://www.cs.columbia.edu/~sedwards/classes/2021/4995-fall/proposals/Palindrome.pdf
+ */
+
+// class PalindromePartitioning {
```
Comment (appgurueu) [inline Recursive/PalindromePartitioning.js]: Please remove this outcommented code.

Comment (Trikcode) [reply Recursive/PalindromePartitioning.js]: I already deleted that in my third commit that i made "clean up code"

I don't know if i am getting you well, but please check out updates of my third commit

Comment (appgurueu) [reply Recursive/PalindromePartitioning.js]: oops yeah sorry, was reviewing the second commit there.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**